lib/logidze/versioned_association.rb

```diff
@@ -0,0 +1,41 @@
+# frozen_string_literal: true
+require 'active_support'
+
+module Logidze
+  module VersionedAssociation
+    module SingularAssociation
+      def reader(force_reload = false)
```

Comment:
That's a lot of information to process :)

1. >And how to handle association loading? I.e. post.at(...).user?
How to prevent double-at!?
All double `at!` wouldn't affect association, because there will be the same timestamp, so `at!` will just return self without any processing. Or I miss some scenario?
2. > we can use stale_target?
What advantage does it have over explicit `reload` or `reset` call?
3. > I found one more problem with collection associations
Got it, I will start with specs for all these methods

Issue gone wild, didn't it? :)